dev/tests/api-functional/testsuite/Magento/GraphQl/OfflineShipping/SetOfflineShippingMethodsOnCartTest.php

```diff
@@ -0,0 +1,209 @@
+    {
+        $this->setShippingMethodAndCheckResponse(
+            'freeshipping',
+            'freeshipping',
+            '0',
+            'Free Shipping - Free'
+        );
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_address_saved.php
+     * @magentoApiDataFixture Magento/Checkout/_files/enable_all_shipping_methods.php
+     */
+    public function testSetUpsOnCart()
+    {
+        $this->setShippingMethodAndCheckResponse(
+            'ups',
+            'GND',
+            '15.61',
+            'United Parcel Service - Ground'
+        );
+    }
+
+    /**
+     * Send request for setting the requested shipping method and check the output
+     *
+     * @param string $shippingCarrierCode
+     * @param string $shippingMethodCode
+     * @param string $shippingAmount
+     * @param string $shippingLabel
+     * @throws \Magento\Framework\Exception\AuthenticationException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function setShippingMethodAndCheckResponse(
+        string $shippingCarrierCode,
+        string $shippingMethodCode,
+        string $shippingAmount,
+        string $shippingLabel
+    ) {
+        $this->quoteResource->load(
+            $this->quote,
+            'test_order_1',
+            'reserved_order_id'
+        );
+        $shippingAddress = $this->quote->getShippingAddress();
+        $shippingAddressId = $shippingAddress->getId();
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
+
+        $query = $this->getQuery(
+            $maskedQuoteId,
+            $shippingMethodCode,
+            $shippingCarrierCode,
+            $shippingAddressId
+        );
+
+        $response = $this->sendRequestWithToken($query);
+
+        $addressesInformation = $response['setShippingMethodsOnCart']['cart']['shipping_addresses'];
+        self::assertEquals($addressesInformation[0]['selected_shipping_method']['carrier_code'], $shippingCarrierCode);
+        self::assertEquals($addressesInformation[0]['selected_shipping_method']['method_code'], $shippingMethodCode);
+        self::assertEquals($addressesInformation[0]['selected_shipping_method']['amount'], $shippingAmount);
+        self::assertEquals($addressesInformation[0]['selected_shipping_method']['label'], $shippingLabel);
+    }
+
+    /**
+     * Generates query for setting the specified shipping method on cart
+     *
+     * @param string $maskedQuoteId
+     * @param string $shippingMethodCode
+     * @param string $shippingCarrierCode
+     * @param string $shippingAddressId
+     * @return string
+     */
+    private function getQuery(
+        string $maskedQuoteId,
+        string $shippingMethodCode,
+        string $shippingCarrierCode,
+        string $shippingAddressId
+    ) : string {
+        return <<<QUERY
+mutation {
+  setShippingMethodsOnCart(input: 
+    {
+      cart_id: "$maskedQuoteId", 
+      shipping_methods: [{
```

Comment:
Need to cover such cases like:
-) Set shipping address twice in the single query